### ISSUE TITLE
Blocked/filtered groups with unread chapters open when using shortcut

### ISSUE THREAD

**Issue by Mnotic** [nekomangaorg/Neko#1813]
### Steps to reproduce

1. Block/filter group with unread chapters
2. Read newest chapter via shortcut

### Expected behavior

Newest unblocked/unfiltered chapter should open

### Actual behavior

Shortcut will open lowest numbered chapter from blocked/filtered group

https://github.com/nekomangaorg/Neko/assets/168328659/01725bfd-3bd1-41fe-8682-513a2f5bc4bc

https://github.com/nekomangaorg/Neko/assets/168328659/29cc0f5b-99a8-4c8c-9288-40f74dbac6ab

### Neko version

2.16.6

### Android version

Android 14

### Device

Samsung Galaxy S23 Ultra

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to the newest version **[Latest](https://github.com/nekomangaorg/Neko/releases/latest)**.
- [X] I have filled out all of the requested information in this form.
